libr/core/panels.c

```diff
@@ -655,26 +659,44 @@ R_API int r_core_visual_panels(RCore *core, RPanels *panels) {
 	key = r_cons_arrow_to_hjkl (okey);
 	r_cons_switchbuf(true);
 	const char *cmd;
+	const char *creg;
 	char buf[128];
 	if (core->print->cur_enabled) {
 		switch (key) {
 			case 9: // TAB
+			case 'J':
+			case 'Z': // SHIFT-TAB
+			case 'K':
 				goto repeat;
 			case 'Q':
 			case 'q':
 				setcursor (core, !core->print->cur_enabled);
 				panels->panel[panels->curnode].refresh = true;
 				goto repeat;
 			case 'i':
-				// insert mode
-				r_line_set_prompt ("insert hex: ");
-				strcpy (buf, "wx ");
-				if (r_cons_fgets (buf + strlen (buf), sizeof (buf) - strlen (buf), 0, NULL) < 0) {
-					buf[0] = '\0';
+				if (!strcmp (panels->panel[panels->curnode].title, PANEL_TITLE_STACK)) {
+					// insert mode
+					r_line_set_prompt ("insert hex: ");
+					strcpy (buf, "wx ");
+					if (r_cons_fgets (buf + strlen (buf), sizeof (buf) - strlen (buf), 0, NULL) < 0) {
+						buf[0] = '\0';
+					}
+					r_core_cmdf (core, strcat (buf, "@ 0x%08" PFMT64x), panels->panel[panels->curnode].addr);
```

Comment:
this can overflow because previous fgets doesnt checks for proper boundaries, using strcat safely is tricky